src/report/annotation.go

```diff
@@ -115,15 +107,15 @@ func sortSeverities(severityCounts map[string]int32) []string {
 }
 
 // sort severity strings by rank, then alphabetically
-func compareSeverities(a, b string) int {
-	rank := rankSeverity(a) - rankSeverity(b)
+func compareSeverities(a, b types.FindingSeverity) int {
+	rank := rankSeverity(string(a)) - rankSeverity(string(b))
```

Comment:
I really appreciate this change here - making sure that it is even a string being used for the `rankSeverity` function in the first place rather than checking in the function if it is a string.

Reply:
@therealvio That's not what's happening. The type of `a` and `b` is `types.FindingSeverity` but the signature of the function is `rankSeverity(string)`. The compiler won't allow this unless the parameter is explicitly converted into a `string`

Reply:
`rankSeverity` is used from contexts where it's a `string` and a `FindingSeverity`, partly due to the vagaries of the AWS API types.

Reply:
Update: no, it's not. It used to be, but not now. I've updated `rankSeverity` to use the enum type.